test/sql/test_iceberg/R/test_iceberg_in_list_predicate

```diff
@@ -0,0 +1,53 @@
+-- name: test_iceberg_in_list_predicate
+create external catalog ice_cat_${uuid0}
+properties
+(
+    "type" = "iceberg",
+    "iceberg.catalog.type" = "hive",
+    "hive.metastore.uris" = "${iceberg_catalog_hive_metastore_uris}"
+);
+set catalog ice_cat_${uuid0};
+create database ice_db_${uuid0};
+use ice_db_${uuid0};
+create table ice_tbl_${uuid0} (
+  col_str string,
+  col_int int
+);
+insert into ice_tbl_${uuid0} values ('1d8cf2a2c0e14fa89d8117792be6eb6f', 2000),
+  ('3e82e36e56718dc4abc1168d21ec91ab', 2000),
+  ('abc', 2000),
+  ('ab1d8cf2a2c0e14fa89d8117792be6eb6f', 2001),
+  ('3e82e36e56718dc4abc1168d21ec91ab', 2001),
+  ('abc', 2001);
+-- result:
+-- !result
+select col_str from ice_tbl_${uuid0} where col_int=2000 and col_str not in (md5('1d8cf2a2c0e14fa89d8117792be6eb6f'), 'abc');
+-- result:
+1d8cf2a2c0e14fa89d8117792be6eb6f
+-- !result
+select col_str from ice_tbl_${uuid0} where col_int=2000 and col_str not in (md5('1d8cf2a2c0e14fa89d8117792be6eb6f'), md5('abc')) order by 1;
+-- result:
+1d8cf2a2c0e14fa89d8117792be6eb6f
+abc
+-- !result
+select col_str from ice_tbl_${uuid0} where col_int=2000 and col_str not in ('1d8cf2a2c0e14fa89d8117792be6eb6f', md5('abc')) order by 1;
+-- result:
+3e82e36e56718dc4abc1168d21ec91ab
+abc
+-- !result
+select col_str from ice_tbl_${uuid0} where col_int=2000 and col_str in (md5('1d8cf2a2c0e14fa89d8117792be6eb6f'), 'abc') order by 1;
+-- result:
+3e82e36e56718dc4abc1168d21ec91ab
+abc
+-- !result
+select col_str from ice_tbl_${uuid0} where col_int=2000 and col_str in (md5('1d8cf2a2c0e14fa89d8117792be6eb6f'), md5('abc'));
+-- result:
+3e82e36e56718dc4abc1168d21ec91ab
+-- !result
+select col_str from ice_tbl_${uuid0} where col_int=2000 and col_str in ('1d8cf2a2c0e14fa89d8117792be6eb6f', md5('abc'));
+-- result:
+1d8cf2a2c0e14fa89d8117792be6eb6f
+-- !result
+drop catalog ice_cat_${uuid0};
```

Comment:
you could drop iceberg table and database you created.

Reply:
done